datadog/resource_datadog_synthetics_test_.go

```diff
@@ -3333,23 +3359,15 @@ func updateSyntheticsAPITestLocalState(d *schema.ResourceData, syntheticsTest *d
 		d.Set("request_proxy", []map[string]interface{}{localProxy})
 	}
 
-	if files, ok := actualRequest.GetFilesOk(); ok && files != nil {
-		localFiles := make([]map[string]interface{}, len(*files))
-		for i, file := range *files {
-			// The file content is kept from the existing localFile from the state,
-			// as the response from the backend contains the bucket key rather than the content.
-			localFile := d.Get(fmt.Sprintf("request_file.%d", i)).(map[string]interface{})
-			localFile["name"] = file.GetName()
-			localFile["original_file_name"] = file.GetOriginalFileName()
-			localFile["type"] = file.GetType()
-			localFile["size"] = file.GetSize()
-
-			if bucket_key, ok := file.GetBucketKeyOk(); ok {
-				localFile["bucket_key"] = bucket_key
-			}
-			localFiles[i] = localFile
+	if files, ok := actualRequest.GetFilesOk(); ok && files != nil && len(*files) > 0 {
+		previousLocalFilesCount := d.Get("request_file.#").(int)
```

Comment:
**nit**: Would you mind replacing all `previous` by `old`? Since we are manipulating arrays, "previous" could mean previous item in the array

It's also the same "convention" that is used elsewhere in the file:
https://github.com/DataDog/terraform-provider-datadog/blob/ca0fa3febf2ba391e465aef444ad7cc6bf3d3892/datadog/resource_datadog_synthetics_test_.go#L51